tests/client_test.py

```diff
@@ -415,7 +415,7 @@ def testDeleteAttachment(self):
         old_rev = doc['_rev']
         db.put_attachment(doc, text_attachment, "test", "text/plain")
         db.delete_attachment(doc, 'test')
-        self.assertRaises(ResourceNotFound, db.fetch_attachment, doc, 'test')
+        self.assertRaises(HTTPError, db.fetch_attachment, doc, 'test')
```

Comment:
Would it be possible to catch `HTTPError` in `fetch_attachment` and re-raise `ResourceNotFound`? Otherwise you're changing the interface of `fetch_attachment`, which will require client code changes.